silkworm/node/db/CMakeLists.txt

```
@@ -0,0 +1,51 @@
+#[[
+   Copyright 2024 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+]]
+
+include("${SILKWORM_MAIN_DIR}/cmake/common/targets.cmake")
+
+add_subdirectory(etl)
+
+find_package(absl REQUIRED)
+find_package(Microsoft.GSL REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(roaring REQUIRED)
+
+set(LIBS_PUBLIC
+    absl::btree
+    absl::flat_hash_map
+    absl::flat_hash_set
+    absl::strings
+    mdbx-static
+    roaring::roaring
+    silkworm_core
+)
+
+# cmake-format: off
+set(LIBS_PRIVATE
+    cborcpp
+    Microsoft.GSL::GSL
+    nlohmann_json::nlohmann_json
+    silkworm_infra
+    silkworm_snapshots
+)
+# cmake-format: on
+
+silkworm_library(
+  silkworm_db
+  PUBLIC ${LIBS_PUBLIC}
+  PRIVATE ${LIBS_PRIVATE}
+  EXCLUDE_REGEX "db/etl"
```

```suggestion
  EXCLUDE_REGEX "etl"
```
not sure about it, can you double check?

---

EXCLUDE_REGEX is a temporary hack to exclude sub-libraries. The plan is to detect them automatically and exclude their subfolders from the sources list. It's on my TODO list ✍️